listen.go

```diff
@@ -37,7 +43,13 @@ func Listen(network, addr string) (net.Listener, error) {
 		return nil, err
 	}
 
-	return &fakeCloseListener{sharedListener: sharedLn.(*sharedListener)}, nil
+	return &fakeCloseListener{sharedListener: sharedLn.(*sharedListener), keepalivePeriod: keepalivePeriod}, nil
+}
+
+// typically applies only to TCP, but using interface for future proofing
+type canSetKeepAlive interface {
+	SetKeepAlivePeriod(d time.Duration) error
+	SetKeepAlive(bool) error
 }
```

Comment:
This is only used on L79 right? I wonder if this should be defined on L78 or thereabouts.

Also, I'm not sure what we're future proofing against. Are you thinking of something I'm not? :sweat_smile: UDP doesn't have connections, and Unix sockets don't need keepalives right?